src/messaging/ExchangeContext.cpp

```diff
@@ -143,7 +143,7 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
 
     {
         // ExchangeContext for group are supposed to always be Initiator
-        if (IsGroupExchangeContext() && !IsInitiator())
+        if (IsGroupExchangeContext() && (!IsInitiator() || GetSessionHandle()->RequireTCP()))
```

Comment:
Please update the comment to explain why this check is what it is.

Because I think this check is wrong.  `GetSessionHandle()->RequireTCP()` is always false, as implemented, for the `IsGroupExchangeContext()` case.

Reply:
Will update the comment. Is RequireTCP() guaranteed to be false in this case, always? The reason I put it was basically to ensure that, i.e, if it is GroupExchangeContext, then it should not try to use TCP.

Reply:
IsGroupExchangeContext() checks for a group session.  Group sessions return false from RequireTCP(), no?